applications/FSIapplication/python_scripts/fsi_analysis.py

```diff
@@ -0,0 +1,338 @@
+            fluid_restart_utility.SaveRestart()
+            structure_restart_utility.SaveRestart()
+
+    def _CreateSolver(self):
+        import python_solvers_wrapper_fsi
+        return python_solvers_wrapper_fsi.CreateSolver(self.model, self.project_parameters)
+
+    def _GetSimulationName(self):
+        fluid_simulation_name = self.project_parameters["fluid_solver_settings"]["problem_data"]["problem_name"].GetString()
+        structure_simulation_name = self.project_parameters["structure_solver_settings"]["problem_data"]["problem_name"].GetString()
+
+        if fluid_simulation_name != structure_simulation_name:
+            err_msg =  'Different fluid and structure simulation names:\n'
+            err_msg += '\t-Fluid simulation name:' + fluid_simulation_name + '\n'
+            err_msg += '\t-Structure simulation name:' + structure_simulation_name + '\n'
+            raise Exception(err_msg)
+
+        return fluid_simulation_name
+
+    def _SetUpListOfProcesses(self):
+        '''
+        Read the definition of initial and boundary conditions for the problem and initialize the processes that will manage them.
+        Also initialize any additional processes present in the problem (such as those used to calculate additional results).
+        '''
+        from process_factory import KratosProcessFactory
+        factory = KratosProcessFactory(self.model)
+        # The list of processes will contain a list with each individual process already constructed (boundary conditions, initial conditions and gravity)
+
+        # Fluid domain processes
+        # Note 1: gravity is constructed first. Outlet process might need its information.
+        # Note 2: initial conditions are constructed before BCs. Otherwise, they may overwrite the BCs information.
+        self._list_of_processes =  factory.ConstructListOfProcesses( self.project_parameters["fluid_solver_settings"]["gravity"] )
+        self._list_of_processes += factory.ConstructListOfProcesses( self.project_parameters["fluid_solver_settings"]["initial_conditions_process_list"] )
+        self._list_of_processes += factory.ConstructListOfProcesses( self.project_parameters["fluid_solver_settings"]["boundary_conditions_process_list"] )
+        self._list_of_processes += factory.ConstructListOfProcesses( self.project_parameters["fluid_solver_settings"]["auxiliar_process_list"] )
+
+        # Structure domain processes
+        self._list_of_processes += factory.ConstructListOfProcesses( self.project_parameters["structure_solver_settings"]["constraints_process_list"] )
+        self._list_of_processes += factory.ConstructListOfProcesses( self.project_parameters["structure_solver_settings"]["loads_process_list"] )
+
+        #TODO this should be generic
+        # Initialize fluid and structure GiD I/O
+        self.fluid_output, self.structure_output = self._SetUpGiDOutput()
+
+        if self.fluid_output is not None:
```

Comment:
If you add this to the list of output_processes then all the calls are taken care of corectly in the baseclass (PrintOutput & IsOutputStep)